### PR TITLE
[TS-250] Implement search functionality for stocks

### PR DESCRIPTION
**Pull request by bowe99**

### CODE REVIEW

**Comment by GadoliniumPotassium** [comment]
Your tests are failing because of "an unexpected use of this". It's failing in the linting stage, I'll look into it to see how you could fix it (unless you already have an idea).

File: src/components/NavBar/NavBar.vue
Comment (GadoliniumPotassium) [inline src/components/NavBar/NavBar.vue:32]: Don't remove this data-cy because otherwise tests will start failing as well as the one on line 36

Comment (bowe99) [reply src/components/NavBar/NavBar.vue:32]: Ok good catch yeah I'm fixing the linting issue I'll add back the data-cy attributes

File: src/components/NavBar/NavBar.vue
Comment (tungleu) [inline src/components/NavBar/NavBar.vue:114]: you will need to change the host as this wont work when it is deployed, would be better if u can use our own existing axios instance perform this API call. take a look at `axios/axios.v1.js` :)

**Review by tungleu — CHANGES_REQUESTED** [review]

File: src/components/NavBar/NavBar.vue
Comment (bowe99) [reply src/components/NavBar/NavBar.vue:114]: Right thanks I'll take a look

Comment (bowe99) [reply src/components/NavBar/NavBar.vue]: Done

File: src/components/NavBar/NavBar.vue
Comment (bowe99) [reply src/components/NavBar/NavBar.vue:32]: Done

**Comment by GadoliniumPotassium** [comment]
I'm going to approve, but I would prefer if you found a way to deal with the warnings before you merge.

**Review by GadoliniumPotassium — APPROVED** [review]
LGTM but we should find a way to show more relevant stocks instead of stocks more by alphabetical order (maybe something we can do later on)

**Comment by GadoliniumPotassium** [comment]
Removing the label tests reduces testing from 85% to 83%

**Review by Ming424 — APPROVED** [review]
image
Looks good to me, but I would say moving the result text align left?

**Review by Ming424 — CHANGES_REQUESTED** [review]
image
image
There is an error inside NavBar, Im checking if its something missing on my side
(update) fixed the node_module

**Review by Ming424 — APPROVED** [review]

File: src/components/NavBar/NavBar.vue
Comment (tungleu) [inline src/components/NavBar/NavBar.vue:117]: remove console log

File: src/components/NavBar/NavBar.vue
Comment (tungleu) [inline src/components/NavBar/NavBar.vue:103]: got this error when I clicked outside the search bar and I also think it's better to use `===` instead of `==` to prevent type coercion 
suggestion: use `?.` operator: `this?.searchQueue.length === 0`
image

File: src/components/NavBar/NavBar.vue
Comment (tungleu) [inline src/components/NavBar/NavBar.vue:104]: `val?.length === 0`

**Review by tungleu — CHANGES_REQUESTED** [review]
Just some minor changes requested, overall the search function works 👍 awesome job

File: src/components/NavBar/NavBar.vue
Comment (bowe99) [reply src/components/NavBar/NavBar.vue:103]: searchQueue is initialized as an empty list so we know it will always work with length

File: src/components/NavBar/NavBar.vue
Comment (bowe99) [reply src/components/NavBar/NavBar.vue:117]: Done

File: src/components/NavBar/NavBar.vue
Comment (bowe99) [reply src/components/NavBar/NavBar.vue:103]: I'll add it because it's extra precaution so it's not a bad idea

File: src/components/NavBar/NavBar.vue
Comment (bowe99) [reply src/components/NavBar/NavBar.vue:104]: Done

**Comment by Ming424** [comment]
@bowe99 I deleted all file related to Label/UserGroup, you can fix that by delete all of them.
There is the folder UserGroupDialog/UserGroupDialog.vue
Also 3 E2E test file as well with their parent folder

**Review by tungleu — APPROVED** [review]

**Review by GadoliniumPotassium — APPROVED** [review]
It's a bit slow but I approve. Good job!

**Pull request merged**
